dataservice/api/read_group/models.py

```diff
@@ -17,17 +20,47 @@ class ReadGroup(db.Model, Base):
 
     external_id = db.Column(db.Text(), nullable=True,
                             doc='Name given to read group by the contributor')
-    paired_end = db.Column(db.Integer(),
-                           doc='The direction of the read')
     flow_cell = db.Column(db.Text(),
                           doc='The identifier of the group\'s flow cell')
     lane_number = db.Column(db.Float(),
                             doc='The group\'s lane')
     quality_scale = db.Column(db.Text(),
                               doc='The scale used to encode quality scores')
-    # NB: Because of the way GenomicFiles may delete themselves,
-    # they do not cascade to the read-group in the ORM, so we explicitly
-    # ON DELETE CASCADE inside Postgres
-    genomic_file_id = db.Column(KfId(), db.ForeignKey('genomic_file.kf_id',
-                                                      ondelete='CASCADE'),
+
+    genomic_files = db.relationship('GenomicFile',
+                                    secondary='read_group_genomic_file',
+                                    backref=db.backref('read_groups'))
+
+
+class ReadGroupGenomicFile(db.Model, Base):
+    """
+    Represents association table between read_group table and
+    genomic_file table. Contains all read_group, genomic_file combiniations.
+    :param kf_id: Unique id given by the Kid's First DCC
+    :param created_at: Time of object creation
+    :param modified_at: Last time of object modification
+    """
+    __tablename__ = 'read_group_genomic_file'
+    __prefix__ = 'RF'
+    __table_args__ = (db.UniqueConstraint('read_group_id',
+                                          'genomic_file_id'),)
+    read_group_id = db.Column(KfId(),
+                              db.ForeignKey('read_group.kf_id'),
+                              nullable=False)
+
+    genomic_file_id = db.Column(KfId(),
+                                db.ForeignKey('genomic_file.kf_id'),
                                 nullable=False)
+    read_group = db.relationship('ReadGroup')
+    genomic_file = db.relationship('GenomicFile')
+
+    def __repr__(self):
+        return "{}-{}".format(self.read_group.external_id,
```

Comment:
Do we use this format anywhere else?
We should probably stay close to the typical `<Object >` format that python uses.
Maybe `<ReadGroupGenomicFile {self.read_group.__repr__()} - {self.genomic_file.__repr__()}>`?

Reply:
Ah sorry, that was there for debugging, let me change it